enterprise/internal/insights/background/historical_enqueuer.go

```diff
@@ -261,64 +273,32 @@ func (h *historicalEnqueuer) buildFrames(ctx context.Context, uniqueSeries map[s
 		return nil // nothing to do.
 	}
 	var multi error
-	for frame := 0; frame < h.framesToBackfill(); frame++ {
-		// Determine the exact start and end time of this timeframe.
-		from := h.now().Add(-time.Duration(frame+1) * h.frameLength())
-		to := h.now().Add(-time.Duration(frame) * h.frameLength())
-		if h.now().After(h.now().Add(24 * time.Hour)) {
-			// We exclude today because it is the regular enqueuer's job to enqueue work for today.
-			to = to.Add(-24 * time.Hour)
-		}
 
-		// Build historical data for this timeframe.
-		softErr, hardErr := h.buildFrame(ctx, uniqueSeries, sortedSeriesIDs, from, to)
-		if softErr != nil {
-			multi = multierror.Append(multi, softErr)
-			continue
-		}
-		if hardErr != nil {
-			return multierror.Append(multi, hardErr)
-		}
-	}
-	return multi // return any soft errors we encountered
-}
+	frames := Frames(h.framesToBackfill(), h.frameLength(), h.now())
 
-// buildFrame is invoked to build historical data for a specific past timeframe that we care about
-// backfilling data for.
-//
-// It is expected to backfill data for all unique series that are missing data, across all repos
-// (using h.allReposIterator.)
-//
-// It may return both hard errors (e.g. DB connection failure, future frames are unlikely to build)
-// and soft errors (e.g. user made a mistake or we did partial work, future frames will likely
-// succeed.)
-func (h *historicalEnqueuer) buildFrame(
-	ctx context.Context,
-	uniqueSeries map[string]*schema.InsightSeries,
-	sortedSeriesIDs []string,
-	from time.Time,
-	to time.Time,
-) (hardErr, softErr error) {
-	// We yield frequently for a small period of time for a few reasons:
-	//
-	// 1. To not call buildSeries too quickly and enqueue millions of jobs rapidly.
-	// 2. To avoid calling repoStore.GetByName() and git.FirstEverCommit() in a tight
-	//    loop for potentially 500,000+ repositories if there is actually no work to
-	//    perform (because all have had historical data built already.)
-	//
+	hardErr := h.allReposIterator(ctx, h.buildForRepo(ctx, uniqueSeries, sortedSeriesIDs, frames, multi))
+	return hardErr
+}
 
-	// For every repository that we want to potentially gather historical data for.
-	hardErr = h.allReposIterator(ctx, func(repoName string) error {
+func (h *historicalEnqueuer) buildForRepo(ctx context.Context, uniqueSeries map[string]*schema.InsightSeries, sortedSeriesIDs []string, frames []compression.Frame, softErr error) func(repoName string) error {
+	return func(repoName string) error {
 		// Lookup the repository (we need its database ID)
 		repo, err := h.repoStore.GetByName(ctx, api.RepoName(repoName))
 		if err != nil {
 			// Ignore RepoNotFoundErr because it could just be that the repository was actually
 			// deleted and allReposIterator had it cached.
 			if _, ok := err.(*database.RepoNotFoundErr); !ok {
 				return err // hard DB error
+			} else {
+				return nil
 			}
 		}
 
+		log15.Info("starting frames", "starting_frames", frames)
```

Comment:
```suggestion
		log15.Info("insights: starting frames", "starting_frames", frames)
```

Make sure to prefix your logs with something, otherwise you're never going to be able to find them